File: CSharp/OpenTraceability/Models/Events/SensorElement.cs
using OpenTraceability.Interfaces;
using OpenTraceability.Utility.Attributes;

namespace OpenTraceability.Models.Events
{
    public class SensorElement
    {
        [OpenTraceabilityObject]
        [OpenTraceability("sensorMetadata", 0)]
        public SensorMetaData MetaData { get; set; }

        [OpenTraceabilityArray]
        [OpenTraceabilityObject]
        [OpenTraceability("sensorReport", 1)]
        public List<SensorReport> Reports { get; set; } = new List<SensorReport>();

        [OpenTraceabilityExtensionElements]
        public List<IEventKDE> ExtensionKDEs { get; internal set; } = new List<IEventKDE>();
    }
}

File: JDK11/opentraceability/src/main/java/opentraceability/models/events/sensorelement.java
package opentraceability.models.events;

import java.util.List;
import opentraceability.interfaces.IEventKDE;
import opentraceability.utility.attributes.OpenTraceabilityArrayAttribute;
import opentraceability.utility.attributes.OpenTraceabilityAttribute;
import opentraceability.utility.attributes.OpenTraceabilityObjectAttribute;
import opentraceability.utility.attributes.OpenTraceabilityExtensionElementsAttribute;

public class SensorElement {
    @OpenTraceabilityObjectAttribute
    @OpenTraceabilityAttribute("", "sensorMetadata", 0)
    public SensorMetaData metaData;

    @OpenTraceabilityArrayAttribute
    @OpenTraceabilityObjectAttribute
    @OpenTraceabilityAttribute("", "sensorReport", 1)
    public List<SensorReport> reports;

    @OpenTraceabilityExtensionElementsAttribute
    public List<IEventKDE> extensionKDEs;
}